ERCS/erc-7585.md

```diff
@@ -0,0 +1,292 @@
+6. The public network can determine the success of the challenge by comparing `proof_result_m` and `proof_result_better_m`. A successful challenge indicates the old storage proof was forged. If no one challenges the published storage proof within a certain timeframe, it can be considered correct from a game-theoretic perspective.
+7. To support healthy competition, the public network should design an appropriate economic model, rewarding users who provide correct storage proofs and penalizing those who submit false ones.
+
+With an understanding of the above process, let us describe the generation and verification of storage proofs more precisely using pseudocode.
+
+```solidity
+function generateProof(mixHash, blockHeight,file) {
+    nonce = getNonce(blockHeight);
+    hash_type = getHashType(mixHash);
+    chunk_hash_array = getChunkHashArray(file,hash_type);
+
+    min_index = 0
+    min_merkle_tree_root = MAX_UINT256
+    min_chunk = None
+
+    m_index = 0;
+    for chunk in file {
+      new_chunk = chunk + nonce;
+      chunk_hash_array[m_index] = getChunkHash(new_chunk,hash_type);
+      merkle_tree_root = getMerkleTreeRoot(chunk_hash_array,hash_type);
+      chunk_hash_array[m_index] = getChunkHash(chunk,hash_type);
+      if (merkle_tree_root < min_merkle_tree_root) {
+        min_merkle_tree_root = merkle_tree_root;
+        min_index = m_index;
+        min_chunk = chunk;
+      }
+      m_index ++;
+    }
+    m_path = getMerkleTreePath(chunk_hash_array, min_index);
+    return strorage_proof(mixHash, blockHeight, min_index, m_path, min_chunk);
+}
+
+function verifyDataProof(mixHash, blockHeight, m_index, m_path, m_leaf_data) {
+    if(current_block_height - blockHeight > MAX_BLOCK_DISTANCE) {
+       revert("proof expired");
+    }
+    hash_type = getHashType(mixHash);
+    merkle_tree_root = getMerkleTreeRootFromPath(m_path,m_leaf_data,hash_type);
+    if(low192(merkle_tree_root) != low192(mixHash)) {
+       revert("invalid proof");
+    }
+
+    nonce = getNonce(blockHeight);
+    proof_result = getMerkleTreeRootFromPath(m_path,m_leaf_data.append(nonce),hash_type);
+    last_proof_result,last_prover = getProofResult(mixHash, blockHeight);
+    if(proof_result < last_proof_result) {
+      emit ProofPunish(last_prover);
+      updateProofResult(mixHash, blockHeight, proof_result, msg.sender);
+    } 
+}
+```
+
+To minimize the size of the storage proof as much as possible, we have optimized the implementation of getMerkleTreeRoot: besides the RootHash, the hash values of other nodes are truncated to the lower 128 bits. This approach effectively compresses the hash value of a complete Merkle tree to half its size. The full implementation details can be found in the subsequent Reference Implementation section.
+
+### Defending Sourcing Attack
```

Comment:
This is a better fit for "Security Consideration"

Reply:
I was also struggling with whether to move this section to "Security Consideration", but considering that defending against Sourcing Attack is usually a core part of constructing proofs for public data storage, I decided to move it to the main text.